### PR TITLE
Fixed #26526 -- Add note to docs on different ordering for QuerySet.latest

### PR DESCRIPTION
**Pull request by g3rd**
Added a note for https://code.djangoproject.com/ticket/26526

### CODE REVIEW

File: docs/ref/models/querysets.txt [hunk 1]
```diff
@@ -1952,6 +1952,16 @@ given parameters.
 Note that ``earliest()`` and ``latest()`` exist purely for convenience and
 readability.
 
+.. admonition:: Note
```
Comment (timgraham) [inline docs/ref/models/querysets.txt]: try to use a more descriptive title than "Note"

File: docs/ref/models/querysets.txt [hunk 2]
```diff
@@ -1952,6 +1952,16 @@ given parameters.
...
+
+    The ordering is delegated to the database's ordering logic. Fields that
+    allow null values are sorted differently across the database
+    vendors.
```
Comment (timgraham) [inline docs/ref/models/querysets.txt]: It might be bout the built-in

File: docs/ref/models/querysets.txt [hunk 3]
```diff
@@ -1952,6 +1952,16 @@ given parameters.
...
+
+    It is possible to filter out null values::
```
Comment (timgraham) [inline docs/ref/models/querysets.txt]: You may want to filter out...

**Comment by timgraham** [comment]
Now that I think about this a bit more, I learn toward having the queryset methods abstract away these database differences by having them filter out null results. Do you have any feelings about this? You could raise it on the django-developers mailing list for further discussion.

**Comment by g3rd** [comment]
I don't have strong feelings about filtering out nulls by default. 

Keeping the logic as is:
- keeps the code simple
- doesn't make assumptions about your data
- doesn't break existing code relying on these methods

Adding a default filter does add to the convenience of the method.

**Comment by tobiasmcnulty** [comment]
@timgraham That seems like a bigger, unnecessary change (admittedly I was convinced by your argument in the original ticket). As you said it seems a bit magical and not necessarily intuitive. I prefer case in which we expose the underlying database discrepancy to the app developer and allow her/him to decide what to do with null values (if they even exist).

**Comment by carljm** [comment]
I agree with @tobiasmcnulty. If we auto-filter nulls, that's a backwards-incompatible change that is difficult for the app developer to reverse if they relied on the null-sorting behavior of their particular database backend. If we document the issue, then it's easy for anyone to filter nulls who wants to do so (including e.g. the author of an app that is supposed to be cross-database).

**Comment by timgraham** [comment]
merged in e6c5e68acd4bff90e371fd148ed6976e147692ad, thanks!

**Pull request closed without merging**